src/App.tsx

```diff
@@ -14,11 +16,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  selectedGoods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    selectedGoods: [],
+  };
+
+  isSelected = (item: string, selectedGoods: string[]) => {
+    if (!this.state.selectedGoods.includes(item)) {
+      this.setState({ selectedGoods: [...selectedGoods, item] });
+    } else {
+      selectedGoods.splice(selectedGoods.indexOf(item), 1);
+      this.setState({ selectedGoods });
+    }
+  };
+
+  setClearList = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  setTitle = (selectedGoods: string[]) => {
+    switch (selectedGoods.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${selectedGoods} is selected`;
+      default: {
+        const title = selectedGoods.map((item) => {
+          if (selectedGoods.indexOf(item) < selectedGoods.length - 2) {
```

Comment:
you can get `index` from the second parameter of callback
```suggestion
        const title = selectedGoods.map((item, index) => {
          if (index < selectedGoods.length - 2) {
```